### DISCUSSION TITLE
'dynamicIO' does not exist in type 'ExperimentalConfig' in NextConfig

### DISCUSSION THREAD

**Discussion by gocallum** [vercel/next.js#72002] [Help]
### What is the documentation issue?

I have tried the following code directly from the example on the documentation 

```typescript
import type { NextConfig } from 'next'
 
const nextConfig: NextConfig = {
  experimental: {
    dynamicIO: true,
  },
}
 
export default nextConfig
```

and I am getting the following compilation error 

Object literal may only specify known properties, and 'dynamicIO' does not exist in type 'ExperimentalConfig'.ts(2353)
config-shared.d.ts(688, 5): The expected type comes from property 'experimental' which is declared here on type 'NextConfig'
(property) dynamicIO: boolean

if I remove the dynamicIO attribute, then, my 'use cache' directive throws error 
 ✓ Compiled /_error in 976ms
 POST / 500 in 7178ms
 ⨯ Error: "use cache" is only available with the experimental.dynamicIO config.

### Is there any context that might help us understand?

$ pnpm next info

Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home
  Available memory (MB): 7832
  Available CPU cores: 8
Binaries:
  Node: 21.7.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.2-canary.7 // Latest available version is detected (15.0.2-canary.7).
  eslint-config-next: 15.0.1
  react: 19.0.0-rc-69d4b800-20241021
  react-dom: 19.0.0-rc-69d4b800-20241021
  typescript: 5.6.3
Next.js Config:
  output: N/A

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/canary/app/api-reference/directives/use-cache

**Comment by samcx** [comment]
@gocallum It is there for me.

Maybe you need to reset the TypeScript server?

![CleanShot 2024-10-28 at 17 20 28@2x](https://github.com/user-attachments/assets/768b7b2b-cb6f-4759-8868-bb25d99d985a)
